app/main/views/suppliers.py

```diff
@@ -67,7 +67,7 @@ def find_suppliers():
     else:
         suppliers_response = data_api_client.find_suppliers(
             name=request.args.get("supplier_name"),
-            duns_number=request.args.get("supplier_duns_number"),
+            duns_number=request.args.get("supplier_duns_number", "").strip(),
```

Comment:
I think this might end up with unexpected behaviour from the API client. Passing through `duns_number=None` to the client means we ignore that parameter completely. Passing through `duns_number=""`... may lead to it filtering for blank DUNSes?

See https://github.com/alphagov/digitalmarketplace-apiclient/blob/master/dmapiclient/data.py#L108

Reply:
Very good shout. I've made the suggested change

Reply:
Cool - are the functional tests happy?

Reply:
Positively elated ^-^